source/served/linters/dscanner.d

```diff
@@ -89,6 +95,23 @@ void lint(Document document)
 	updateDiagnostics(document.uri);
 }
 
+bool isExcludedFile(DocumentUri uri, string[] excludedFiles)
+{
+	auto workspace = workspace(uri);
+
+	if (!uri.startsWith(workspace.folder.uri))
+		return false;
+
+	auto file = uri[workspace.folder.uri.length..$].stripLeft("/");
```

Comment:
```suggestion
	auto file = uriToFile(uri);
```

Reply:
The change would break the feature, as the uri is always absolute, but the exclude rules / globs are relative to the workspace folder. Thus we need to relativize the uri, which is exactly what the code is doing here. I could switch it to using [std.path.relativePath](https://dlang.org/library/std/path/relative_path.html) if thats better.

Reply:
ah right sorry, didn't realize what exactly this doing, I just saw `auto file = uri` and kinda immediately thought the code must be converting uri to file manually without actually reading it